### ISSUE TITLE
Procedural Modules - Stock part highlighting works inconsistently

### ISSUE THREAD

**Issue by shadowmage45** [shadowmage45/SSTULabs#10]
Likely due to constantly re-building the model.  Need to check procedural parts / etc to see if they handle this, and how.

Might try just deleting and recreating the highlighter whenever part model is rebuilt.

**Comment by shadowmage45** [comment]
Highlighting works on a re-loaded part, or revert-to-editor part.  Should definitely investigate recreating the highlighter entirely after meshes are updated.

**Comment by shadowmage45** [comment]
More research -- might just need to recreate the parts' highlighter component whenever the model meshes are changed or updated.
Could be as simple as:

```
part.highligher = new Highlighter(part);
```

**Comment by shadowmage45** [comment]
Solved....
````
        public static void updatePartHighlighting(Part part)
        {
            Component c = part.gameObject.GetComponent<HighlightingSystem.Highlighter>();
            if (c != null) { Component.DestroyImmediate(c); }
            part.highlighter = part.gameObject.AddComponent<HighlightingSystem.Highlighter>();
        }
````
Will be available next update.  Fixes procedural decoupler, modular fuel tanks, modular upper stage, interstage petal adapter, and node-fairings

**Comment by shadowmage45** [comment]
This is.... potentially unsolvable without some patches to stock code.  It does some strange caching of the highlighter somewhere that I do not have access to.  Have opened up a forum thread to see if I can get any more information / if anyone else has any knowledge about it.
http://forum.kerbalspaceprogram.com/index.php?/topic/129527-re-initialize-editor-part-highlighting/

**Comment by shadowmage45** [comment]
Have finally tracked down the problem -- a private cached list of renderers stored in Part, that is only updated/initialized when the first mouse-over event is called.

Have begged NathanKell to look into fixing it, and we'll hope that such a fix or workaround becomes available with 1.1.

**Comment by shadowmage45** [comment]
Have a WIP workaround, hoping to get it deployed and tested tonight; should be available with the next update.  (Not a proper solution, but a hack-around like I have done with the black-body glow...less than optimal, but good enough for these purposes, esp. since it will only be active in the editor).
